### ISSUE TITLE
TypeError: cafile must be None or a byte string - AGAIN lol

### ISSUE THREAD

**Issue by whipcracker** [pymedusa/Medusa#1422]
### Before submitting your issue:

Enable debug logging in Medusa settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit: Custom develop branch with SSL fixes left out (ask @fernandog)
OS: Debian
What you did: Nothing... not this time anyway.
What happened: Error msg popped up as a warning.
What you expected: Nothing, really.
Logs:
```
2016-11-01 03:00:02 ERROR    SHOWUPDATER :: [e372494] Exception generated in thread SHOWUPDATER: cafile must be None or a byte string
Traceback (most recent call last):
  File "/media/sdq1/home/**********/.sickrage/medusa/scheduler.py", line 103, in run
    self.action.run(self.force)
  File "/media/sdq1/home/**********/.sickrage/medusa/show_updater.py", line 68, in run
    t = app.indexerApi(indexer).indexer(**indexer_api_params)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/indexer_api.py", line 34, in indexer
    return indexerConfig[self.indexer_id]['module'](*args, **kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/tvdbv2/tvdbv2_api.py", line 66, in __init__
    access_token = auth_api.login_post(authentication_string)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/apis/authentication_api.py", line 124, in login_post
    callback=params.get('callback'))
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 324, in call_api
    response_type, auth_settings, callback)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 150, in __call_api
    post_params=post_params, body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 360, in request
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 208, in POST
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 143, in request
    headers=headers)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 73, in request
    **urlopen_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 151, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/poolmanager.py", line 165, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 351, in _make_request
    self._validate_conn(conn)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 814, in _validate_conn
    conn.connect()
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connection.py", line 289, in connect
    ssl_version=resolved_ssl_version)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 328, in ssl_wrap_socket
    ctx.load_verify_locations(ca_certs, ca_cert_dir)
  File "/usr/lib/python2.7/dist-packages/OpenSSL/SSL.py", line 303, in load_verify_locations
    raise TypeError("cafile must be None or a byte string")
TypeError: cafile must be None or a byte string
```

**Comment by whipcracker** [comment]
Would most likely be related to https://github.com/pymedusa/Medusa/issues/1385

**Comment by whipcracker** [comment]
Gonna tag @fernandog and @ratoaq2 due to their help previously to investamagate.

**Comment by whipcracker** [comment]
Can confirm I get this issue when I try to edit a show (in this case, "Timeless (2016)" and adding a scene exception as "Timeless" so that Medusa can snatch the correct torrent) and get this in the logs:

````
2016-11-01 05:14:59 ERROR    Thread-27 :: [e372494] Exception generated in thread HTTP REQUEST: cafile must be None or a byte string
Traceback (most recent call last):
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/core/base.py", line 271, in async_call
    result = function(**kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/home/handler.py", line 1298, in editShow
    if indexerLang and indexerLang in app.indexerApi(show_obj.indexer).indexer().config['valid_languages']:
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/indexer_api.py", line 34, in indexer
    return indexerConfig[self.indexer_id]['module'](*args, **kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/tvdbv2/tvdbv2_api.py", line 66, in __init__
    access_token = auth_api.login_post(authentication_string)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/apis/authentication_api.py", line 124, in login_post
    callback=params.get('callback'))
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 324, in call_api
    response_type, auth_settings, callback)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 150, in __call_api
    post_params=post_params, body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 360, in request
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 208, in POST
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 143, in request
    headers=headers)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 73, in request
    **urlopen_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 151, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/poolmanager.py", line 165, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 351, in _make_request
    self._validate_conn(conn)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 814, in _validate_conn
    conn.connect()
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connection.py", line 289, in connect
    ssl_version=resolved_ssl_version)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 328, in ssl_wrap_socket
    ctx.load_verify_locations(ca_certs, ca_cert_dir)
  File "/usr/lib/python2.7/dist-packages/OpenSSL/SSL.py", line 303, in load_verify_locations
    raise TypeError("cafile must be None or a byte string")
TypeError: cafile must be None or a byte string
2016-11-01 05:14:10 DEBUG    CHECKVERSION :: [e372494] check_for_new_news: Checking GitHub for latest news.
2016-11-01 05:14:10 INFO     CHECKVERSION :: [e372494] Version checking is disabled, not checking for the newest version
2016-11-01 05:14:10 DEBUG    CHECKVERSION :: [e372494] git ls-remote --heads origin : returned successful
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] Executing git ls-remote --heads origin with your shell in /media/sdq1/home/**********/.sickrage
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] git symbolic-ref -q HEAD : returned successful
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] Executing git symbolic-ref -q HEAD with your shell in /media/sdq1/home/**********/.sickrage
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] git config remote.origin.pushurl https://github.com/**********/Medusa.git : returned successful
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] Executing git config remote.origin.pushurl https://github.com/**********/Medusa.git with your shell in /media/sdq1/home/**********/.sickrage
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] git config remote.origin.url https://github.com/pymedusa/Medusa.git : returned successful
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] Executing git config remote.origin.url https://github.com/pymedusa/Medusa.git with your shell in /media/sdq1/home/**********/.sickrage
2016-11-01 05:14:09 DEBUG    CHECKVERSION :: [e372494] Starting new thread: CHECKVERSION
2016-11-01 05:14:03 ERROR    Thread-26 :: [e372494] Exception generated in thread HTTP REQUEST: cafile must be None or a byte string
Traceback (most recent call last):
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/core/base.py", line 271, in async_call
    result = function(**kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/home/handler.py", line 1298, in editShow
    if indexerLang and indexerLang in app.indexerApi(show_obj.indexer).indexer().config['valid_languages']:
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/indexer_api.py", line 34, in indexer
    return indexerConfig[self.indexer_id]['module'](*args, **kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/tvdbv2/tvdbv2_api.py", line 66, in __init__
    access_token = auth_api.login_post(authentication_string)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/apis/authentication_api.py", line 124, in login_post
    callback=params.get('callback'))
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 324, in call_api
    response_type, auth_settings, callback)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 150, in __call_api
    post_params=post_params, body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 360, in request
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 208, in POST
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 143, in request
    headers=headers)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 73, in request
    **urlopen_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 151, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/poolmanager.py", line 165, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 351, in _make_request
    self._validate_conn(conn)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 814, in _validate_conn
    conn.connect()
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connection.py", line 289, in connect
    ssl_version=resolved_ssl_version)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 328, in ssl_wrap_socket
    ctx.load_verify_locations(ca_certs, ca_cert_dir)
  File "/usr/lib/python2.7/dist-packages/OpenSSL/SSL.py", line 303, in load_verify_locations
    raise TypeError("cafile must be None or a byte string")
TypeError: cafile must be None or a byte string
2016-11-01 05:12:59 ERROR    Thread-22 :: [e372494] Exception generated in thread HTTP REQUEST: cafile must be None or a byte string
Traceback (most recent call last):
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/core/base.py", line 271, in async_call
    result = function(**kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/home/handler.py", line 1298, in editShow
    if indexerLang and indexerLang in app.indexerApi(show_obj.indexer).indexer().config['valid_languages']:
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/indexer_api.py", line 34, in indexer
    return indexerConfig[self.indexer_id]['module'](*args, **kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/tvdbv2/tvdbv2_api.py", line 66, in __init__
    access_token = auth_api.login_post(authentication_string)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/apis/authentication_api.py", line 124, in login_post
    callback=params.get('callback'))
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 324, in call_api
    response_type, auth_settings, callback)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 150, in __call_api
    post_params=post_params, body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 360, in request
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 208, in POST
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 143, in request
    headers=headers)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 73, in request
    **urlopen_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 151, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/poolmanager.py", line 165, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 351, in _make_request
    self._validate_conn(conn)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 814, in _validate_conn
    conn.connect()
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connection.py", line 289, in connect
    ssl_version=resolved_ssl_version)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 328, in ssl_wrap_socket
    ctx.load_verify_locations(ca_certs, ca_cert_dir)
  File "/usr/lib/python2.7/dist-packages/OpenSSL/SSL.py", line 303, in load_verify_locations
    raise TypeError("cafile must be None or a byte string")
TypeError: cafile must be None or a byte string
````

**Comment by whipcracker** [comment]
Can't seem to add a new show either... unsure if it's related to this issue or not, but The TVDBv2 times out almost instantly when searching for the show.

**Comment by ratoaq2** [comment]
Are you able to test this branch: `bugfix/pyopenssl-issue-209`  ?

**Comment by whipcracker** [comment]
I'm gonna come back at you with a noob question.. 

What's the command lines to make this happen? Also, I've had to change the config.ini and edit ssl_verify to 0. Would I need to change this back first as well? 

Cheers!

**Comment by ratoaq2** [comment]
```
git checkout bugfix/pyopenssl-issue-209
git pull
git reset --hard origin/bugfix/pyopenssl-issue-209
git clean -d -f lib
```
Just to be safe


And you can turn on ssl back so you can test if everything works.

**Comment by whipcracker** [comment]
Tried "git checkout bugfix/pyopenssl-issue-209" and got
error: pathspec 'bugfix/pyopenssl-issue-209' did not match any file(s) known to git.

**Comment by ratoaq2** [comment]
git fetch origin 


Sorry

**Comment by whipcracker** [comment]
Just on the branch now, will try some manual searches and see how I go.

**Comment by whipcracker** [comment]
Hi Rato,

Can't edit a show, still can't add one either. As soon as I try to add the scene exception, it just sits with a blank screen and the url ending in home/editShow

Let me know what you'd like to do?

**Comment by whipcracker** [comment]
To expand on that, editing the show gave me this in the Error log:

````
2016-11-01 08:03:25 ERROR    Thread-26 :: [e372494] Exception generated in thread HTTP REQUEST: cafile must be None or a byte string
Traceback (most recent call last):
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/core/base.py", line 271, in async_call
    result = function(**kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/home/handler.py", line 1298, in editShow
    if indexerLang and indexerLang in app.indexerApi(show_obj.indexer).indexer().config['valid_languages']:
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/indexer_api.py", line 34, in indexer
    return indexerConfig[self.indexer_id]['module'](*args, **kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/tvdbv2/tvdbv2_api.py", line 66, in __init__
    access_token = auth_api.login_post(authentication_string)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/apis/authentication_api.py", line 124, in login_post
    callback=params.get('callback'))
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 324, in call_api
    response_type, auth_settings, callback)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 150, in __call_api
    post_params=post_params, body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 360, in request
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 208, in POST
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 143, in request
    headers=headers)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 73, in request
    **urlopen_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 151, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/poolmanager.py", line 165, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 351, in _make_request
    self._validate_conn(conn)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 814, in _validate_conn
    conn.connect()
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connection.py", line 289, in connect
    ssl_version=resolved_ssl_version)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 328, in ssl_wrap_socket
    ctx.load_verify_locations(ca_certs, ca_cert_dir)
  File "/usr/lib/python2.7/dist-packages/OpenSSL/SSL.py", line 303, in load_verify_locations
    raise TypeError("cafile must be None or a byte string")
TypeError: cafile must be None or a byte string
````

**Comment by ratoaq2** [comment]
Fixed by #1436  in develop branch. Please reopen of the issue persists
